code/datums/uplink/announcements.dm

```diff
@@ -1,6 +1,8 @@
 /****************
 * Announcements *
 *****************/
+var/cooldown = 0
+var/cooldownr = 0
```

Comment:
Use static variables. Specially with names as generic as these.

Put this inbetween lines 91 and 92.
var/static/cooldown

And this inbetween lines 107 and 108.
var/static/cooldownr (technically, you can reuse the same name now. So they can both be called cooldown and not interfere with eachother.)